src/pipeline.cpp

```diff
@@ -138,32 +152,27 @@ static PJ_LP     pipeline_reverse (PJ_XY xy, PJ *P);
 
 void pj_pipeline_assign_context_to_steps( PJ* P, PJ_CONTEXT* ctx )
 {
-    for (int i = 1; i <= static_cast<struct pj_opaque*>(P->opaque)->steps; i++)
-        proj_assign_context(static_cast<struct pj_opaque*>(P->opaque)->pipeline[i], ctx);
+    auto opaque = static_cast<struct pj_opaque*>(P->opaque);
+    for( auto& step: opaque->steps )
+        proj_assign_context(step.pj, ctx);
 }
 
 
 static PJ_COORD pipeline_forward_4d (PJ_COORD point, PJ *P) {
-    int i, first_step, last_step;
-
-    first_step = 1;
-    last_step  = static_cast<struct pj_opaque*>(P->opaque)->steps + 1;
-
-    for (i = first_step;  i != last_step;  i++)
-        point = proj_trans (static_cast<struct pj_opaque*>(P->opaque)->pipeline[i], PJ_FWD, point);
+    auto opaque = static_cast<struct pj_opaque*>(P->opaque);
```

Comment:
I think this would be even more readable if you call the variable "pipeline" instead of "opaque"

Applicable several places below.

Reply:
done